eyeshade/workers/referrals.js

```diff
@@ -103,24 +108,30 @@ module.exports.consumer = (runtime) => {
 
       const normalizedChannel = normalizeChannel(referral._id.publisher)
       const id = transaction.id.referral(txId, normalizedChannel)
-      if (inserting[id]) {
+      return {
+        id,
+        referral
+      }
+    })
+    const ids = docs.map(({ id }) => id)
+    const {
+      rows: previouslyInserted
+    } = await postgres.query(getTransactionsById, [ids])
+    return Promise.all(docs.map(async ({ id: targetId, referral }) => {
```

Comment:
should we be `await`ing this instead of returning?

Reply:
it ends up being the same thing because we are `await` ing on the other side and this is an `async` function